internal/provider/resource_application_test.go

```diff
@@ -203,6 +203,74 @@ func TestAcc_CharmUpdates(t *testing.T) {
 	})
 }
 
+func TestAcc_ResourceRevisionUpdatesLXD(t *testing.T) {
+	if testingCloud != LXDCloudTesting {
+		t.Skip(t.Name() + " only runs with LXD")
+	}
+	modelName := acctest.RandomWithPrefix("tf-test-resource-revision-updates-lxd")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { testAccPreCheck(t) },
+		ProtoV6ProviderFactories: frameworkProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccResourceApplicationWithRevisionAndConfig(modelName, "juju-qa-test", 21, "", "foo-file", 4),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("juju_application.juju-qa-test", "resources.foo-file", "4"),
+				),
+			},
+			{
+				// change resource revision to 3
+				Config: testAccResourceApplicationWithRevisionAndConfig(modelName, "juju-qa-test", 21, "", "foo-file", 3),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("juju_application.juju-qa-test", "resources.foo-file", "3"),
+				),
+			},
+			{
+				// change back to 4
+				Config: testAccResourceApplicationWithRevisionAndConfig(modelName, "juju-qa-test", 21, "", "foo-file", 4),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("juju_application.juju-qa-test", "resources.foo-file", "4"),
+				),
```

Comment:
There is another scenario to cover to replace this. Removing the resource revision of the plan, updating the charm to a new channel, while removing the revision.

Reply:
added two tests:

1. starts with no resource revision in the plan, then adds it
2. starts with resource revision in the plan, then removes it

but.. the issue is that i cannot assert what juju is given in the update command, just that the state does not contain a resource revision definition..

Reply:
we're working to add more unit tests rather than only the acceptance tests.

Reply:
The Acceptance tests also run Read after each step and compare.